src/core/path.c

```diff
@@ -532,17 +536,24 @@ static void path_enter_running(Path *p) {
         path_enter_dead(p, PATH_FAILURE_RESOURCES);
 }
 
-static bool path_check_good(Path *p, bool initial, bool from_trigger_notify) {
+static int path_check_good(Path *p, bool initial, bool from_trigger_notify, char **ret_trigger_path) {
         assert(p);
+        assert(ret_trigger_path);
 
         LIST_FOREACH(spec, s, p->specs)
-                if (path_spec_check_good(s, initial, from_trigger_notify))
-                        return true;
+                if (path_spec_check_good(s, initial, from_trigger_notify)) {
+                        *ret_trigger_path = strdup(s->path);
```

Comment:
as per coding style: do not clobber return values on failure. Here, if the allocation fails, you will set the returb parameter to NULL...